Feature/update tests

elgator opened the pull request:
Hi,
A PR to add a number of unit-tests:

- Tests for context managing mechanics: context metaclass, GameEntity.
- Basic tests for API calls. 
- Tests for Command classes. PassCommand is not tested because it now performs some magic.
- A test for ConsoleUI and GameController that reads commands from a file.

elgator committed:
Add basic API tests

elgator committed:
Add context tests

elgator committed:
Add tests for Command and MoveCommand

elgator committed:
Merge remote-tracking branch 'origin/develop' into feature/update-tests

elgator committed:
Fix tests broken by name changes in develop

elgator committed:
Add tests for Commands (except PassCommand)

elgator committed:
Add tests for search_context and GameEntity

elgator committed:
Add a basic test for GameController and ConsoleUI

almaslov commented on test/test_core/test_context.py:
```diff
@@ -0,0 +1,69 @@
+        self.assertNotEqual(ctx_id_1, ctx_id_2)
+
+
+class MockController(metaclass=ContextRegistrationMeta,
```
Since ContextRegistrationMeta is the code we try to test, we could wrap MockController class creation in a function. Something like:
```
def construct_controller_mock_class():
    class MockController: 
        ...

    return MockController
```
This approach helps to catch exceptions from ContextRegistrationMeta while running a test case, not while parsing the "test_context.py" source file. If such an exception appears, the test log will be more clear.

elgator replied:
A valid point. I concluded that existence of ContextRegistrationMeta is guaranteed by addressing ._ctx property.
I tried to include this definition into setUp(), but something went wrong.

almaslov commented on test/test_core/test_game_entity.py:
```diff
@@ -0,0 +1,49 @@
+    def test_assert_has_context(self):
+        with self.assertRaises(ContextError):
+            with warnings.catch_warnings(): # catch does not work
+                warnings.simplefilter('ignore')
```
Which warnings suppressed here?
I couldn't find any in the project code.

elgator replied:
GameEntity is created here without context, so GameEntityCreationMeta logs a warning. It appears only when I run test only in this file. When I run the whole suit it shows nothing. Perhaps this is so because it is a log, not throw.

almaslov reviewed (commented):
Hello!
Thanks for your PR.

elgator committed:
Restructure tests for GameEntity and ContextRegistration

Move MockController class declaration into test class
Remove warnings filtering

elgator committed:
Add tests for PassCommand and empty outbreak stack

D-Mantrid committed:
Updated tests for `core.game_entity` and `core.api` module.

Provided test case for `GameEntityCreationMeta` and simplified test case for `GameEntity`.
Refined codestyle for APITestCase.

D-Mantrid committed:
Provided test case for issue #18.

Provided test case to check that game outbreaks occur properly during the infection phase.

D-Mantrid committed:
Expanded and simplified `core.context` test case.

Got rid of unnecessary mentions of game controller and simplified mocks.
Split test methods for module-level functions.
Provided comprehensive test case for metaclass.

elgator committed:
Add some tests for GameController

Namely, for start_game and send

D-Mantrid committed:
A bit more tests for GameController and context.py.

D-Mantrid reviewed (approved):
We have enough new tests here.
New tests will be included in the next iteration.
